config/config.go

```diff
@@ -0,0 +1,176 @@
+	if err != nil {
+		return nil, err
+	}
+	return &config, nil
+}
+
+// initConfig gathers the configuration values from command line and config file (if provided) and returns
+// them in a ConfValues struct. If both are provided, the command line flags take precedence.
+func initConfig() (ConfValues, error) {
+	fs := pflag.NewFlagSet("nav", pflag.ContinueOnError)
+	var confValues ConfValues
+	var configPath string
+
+	err := parseCommandLine(fs, &configPath, os.Args)
+	if err != nil {
+		return ConfValues{}, fmt.Errorf("error initializing command line flags: %w", err)
+	}
+
+	if len(os.Args) == 1 {
+		return ConfValues{}, fmt.Errorf("error: no configuration was specified, please specify a configuration file or use the command line flags; " +
+			"use -h or --help for more information")
+	}
+
+	if configPath != "" {
+		err = loadConfigFile(&confValues, configPath)
+		if err != nil {
+			return ConfValues{}, fmt.Errorf("error: %w", err)
+		}
+	}
+
+	setFlags(fs, &confValues)
+	if err := confValues.validate(); err != nil {
+		return ConfValues{}, fmt.Errorf("invalid configuration: %w", err)
+	}
+
+	return confValues, nil
+}
+
+func loadConfigFile(cfg *ConfValues, configPath string) error {
+	cleanedPath := filepath.Clean(configPath)
+
+	if cleanedPath != configPath {
+		return fmt.Errorf("invalid file path")
+	}
+
+	jsonFile, err := os.Open(cleanedPath)
+	if err != nil {
+		return fmt.Errorf("problem while opening config file: %w", err)
+	}
+	defer func() {
+		closeErr := jsonFile.Close()
+		if err == nil {
+			err = closeErr
+		}
+	}()
+
+	byteValue, _ := io.ReadAll(jsonFile)
+	err = json.Unmarshal(byteValue, cfg)
+	if err != nil {
+		return fmt.Errorf("problem while parsing config file: %w", err)
+	}
+	return nil
+}
+
+func (cfg *ConfValues) validate() error {
+	if cfg.Symbol == "" {
+		return fmt.Errorf("symbol must be specified")
+	}
+	if cfg.MaxDepth < 0 {
+		return fmt.Errorf("invalid depth: %d", cfg.MaxDepth)
+	}
+	if err := validateDBInstance(&cfg.DBInstance); err != nil {
+		return err
+	}
+	if err := validateDBDriver(&cfg.DBDriver); err != nil {
+		return err
+	}
+	if err := validateMode(&cfg.Mode); err != nil {
+		return err
+	}
+	if err := validateType(&cfg.Type); err != nil {
+		return err
+	}
+
+	return nil
+}
+
+func validateDBInstance(i *int) error {
+	if *i < 0 {
+		return fmt.Errorf("invalid database instance: %d", *i)
+	}
+	switch *i {
+	case 0:
+		*i = defaultDBInstance
+		fmt.Printf("No database instance specified. Defaulting to %d.\n", defaultDBInstance)
+		return nil
+	default:
+		return nil
+	}
+}
+
+func validateDBDriver(d *string) error {
+	switch *d {
+	case "":
+		*d = defaultDBDriver
+		fmt.Printf("No database driver specified. Defaulting to %s.\n", defaultDBDriver)
+		return nil
+	case "mysql", "postgres", "sqlite3":
+		return nil
+	default:
+		return fmt.Errorf("invalid database driver: %s\nChoose one of the following: mysql, postgres or sqlite3", *d)
+	}
+}
+
+func validateMode(m *c.OutMode) error {
```

Comment:
So far, the mode argument was a numerical value while the output type was a string. 
Reviewing the repository history, it appears that the mode was implemented prior to the output type. 
When I introduced the output type, I believed that having too many numerical values would make the command line interface confusing, so I opted for a string representation instead to improve readability. 
While it wasn't an urgent matter at the time, I always had the notion that the mode would eventually need to be converted into a string argument as well. 
With our current efforts focused on refactoring the configuration-related aspects, I believe it is an appropriate time to make this conversion.

Reply:
I think this is also something that can be written down as a Github issue and addressed in a future PR, since I'd like to avoid changing this PR much and complicating its review.

Reply:
Created an issue for it, closing this conversation